### PR TITLE
Implementation for Assigning Tasks to Tags + Bug Reports Improvements + Fix for Due Date Sync

### LINKED ISSUE #409: Skills screen says "Unlock at lvl 10" instead of level 11
When a low-level player uses the app, the Skills screen is unavailable with a message saying that skills unlock at level 10. It should say level 11.

We see a few questions in the Tavern from players who think there's a bug because they can't get to the skills screen at level 10.

### PR DESCRIPTION
**Pull request by WhatsEmo**
- [x] #361 
- [x] #409 
- [x] #336 
- [x] #417 

Fixed #409's visual glitch:
-
 -Changed "Unlock at lvl 10" -> "Unlock at lvl 11"
Implemented Tag Assignments to Tasks (#361)
- 
- Check Boxes available for users to select Tags
- Users can now edit old Tasks and add Tags
- Check Boxes are pre-checked if user has assigned Tags to a Task
- All information of the Tags are saved sync'd to the SQL DB. -> Users can see all their adjustments both on the Web and Android.

Improved Bug Report Template Emails (#336)
-
- Added getUserID() function to MainActivity.java
- Passed UserID through Intents ( MainActivity -> MainDrawerBuilder -> AboutActivity -> AboutFragment)
- Modified sendEmail(String subject) in AboutFragment.java to add a
template email for Bug Reports

Due dates of to-dos are now Synced/Saved #417
-
- Added a check to add all due dates the user submits.
- Will add something to allow user to decide if they want to add a due date or not.
   - Currently Due Dates are always assigned.

Possible Improvements:
-
For #361:
- Possible implementation of a HashMap which will improve time and space complexities.
- I could work on the above, but I think I should work on other features.

For #417:
- Add something to let user decide if they want to add a due date

### CODE REVIEW

File: Habitica/res/layout/activity_task_form.xml [hunk 1]
```diff
@@ -283,6 +283,29 @@
         </LinearLayout>
 
         <LinearLayout
```
Comment (WhatsEmo) [inline Habitica/res/layout/activity_task_form.xml]: Changed XML to add the "Tags" Category on the TaskFormActivity

File: Habitica/res/values/strings.xml [hunk 2]
```diff
@@ -120,7 +121,7 @@
     <string name="coming_soon">Coming Soon</string>
     <string name="chat_flag_confirmation">Are you sure you want to report this message for violation?</string>
     <string name="flag_confirm">Flag Message</string>
-    <string name="unlock_lvl_10">Unlock at lvl 10</string>
+    <string name="unlock_lvl_10">Unlock at lvl 11</string>
```
Comment (WhatsEmo) [inline Habitica/res/values/strings.xml]: Change to fix visual glitch

File: Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java [hunk 3]
```diff
@@ -268,6 +299,34 @@ public void onClick(View v) {
         });
     }
 
+
```
Comment (WhatsEmo) [inline Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java]: Populates the Task Form with Check Boxes that allow users to select Tags

File: Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java [hunk 4]
```diff
@@ -366,6 +425,15 @@ private void populate(Task task) {
         taskNotes.setText(task.notes);
         taskValue.setText(String.format("%.0f", task.value));
```
Comment (WhatsEmo) [inline Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java]: Helps with visually loading in the Tags that were set by the user for a specific Task

File: Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java [hunk 5]
```diff
@@ -433,6 +501,28 @@ private boolean saveTask(Task task) {
 
         task.notes = taskNotes.getText().toString();
 
+
```
Comment (WhatsEmo) [inline Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java]: Save their newly created/updated Task with their associated Tags

File: Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java [hunk 6]
```diff
@@ -510,7 +600,7 @@ private void prepareSave() {
             this.task.save();
             new Select()
                     .from(Tag.class)
-                    .where(Condition.column("id").in("", tags.toArray())).async().queryList(tagsSearchingListener);
+                    .where(Condition.column("id").in("", userSelectedTagIds.toArray())).async().queryList(tagsSearchingListener);
```
Comment (WhatsEmo) [inline Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java]: Only add the Tags the user specified.

tags = List of all tags
userSelectedTagsId = list of tags user selected

**Comment by vIiRuS** [comment]
I'm not sure why, but it crashes for me after saving a task. In TasksFragment.java at line 165 it tries to access `user.getTags()`, but the user object is null.

Can you also add a checkbox to enable/disable duedates for tasks as suggested in possible improvements?

Other than those two things, this looks good!
While it's fine for this time, can you make multiple PRs in the future separated by features/bugfixes?

**Comment by WhatsEmo** [comment]
Yeah I was working on that the past couple of days and implemented a checkbox, but there's a problem I can't see to solve. I can send a Pull request for the checkbox update, but I don't know how to update the info to the server.

Currently this is what happens:
-> User makes a new Task (without Due Date)
-> Server updates properly
-> User wants to make changes to Task to give a Due Date
-> Server updates properly
-> User wants to remove Due Date
-> Server doesn't make the Due Date "null"

**Comment by vIiRuS** [comment]
Are you sending the string null? or a null object?

**Comment by WhatsEmo** [comment]
I did task.setDueDate(null) so that should be sending a null object. I can't seem to find the place where the app tells the server that the user is updating the duedate column of the database.

**Comment by vIiRuS** [comment]
Just tried to take a look and noticed that you accidentally commited several merge conflict stuff. (search for "<<<<<<< HEAD" ). Could you resolve those?

**Comment by WhatsEmo** [comment]
Finished clearing the merge conflicts. Sorry it took awhile! I'm fairly new to GitHub and I'm still getting the hang of things. Thanks.

Any other problems with this Pull request?

**Comment by vIiRuS** [comment]
Added serializeNulls() to the gson config, so that empty duedates are set.
Thanks for your commits!

Can you also tell me your habitica user id, so that I can note this as a contribution?

**Comment by WhatsEmo** [comment]
Habitica ID:ec0c55e5-36f3-4389-b67c-b264bc7885c0

**Pull request merged**